### ISSUE TITLE
invoker shutdown does not take runc into account

### ISSUE THREAD

**Issue by domdom82** [openwhisk/openwhisk#1982]
## Steps to reproduce the issue:

1. install whisk  
2. create a new action and run it
3.  `docker stop invoker0`

## Provide the expected results and outputs:

```
Shutdown hook should unpause (runc resume) the container and remove it.
```

## Provide the actual results and outputs:

```
[2017-03-08T09:33:36.374Z] [WARN] [??] [ContainerPool] Shutdown hook activated.  Starting container shutdown
[2017-03-08T09:33:36.374Z] [INFO] [#sid_101] [ContainerUtils] [marker:invoker_docker.ps_start:23342545]
[2017-03-08T09:33:36.374Z] [INFO] [#sid_101] [SimpleExec] Running command: /usr/bin/docker ps --no-trunc -a
[2017-03-08T09:33:36.391Z] [INFO] [#sid_101] [SimpleExec] Done running command: /usr/bin/docker ps --no-trunc -a
[2017-03-08T09:33:36.391Z] [INFO] [#sid_101] [ContainerUtils] [marker:invoker_docker.ps_finish:23342564:19]
[2017-03-08T09:33:36.391Z] [INFO] [#sid_101] [ContainerPool] killStragglers now removing 1 leftover containers
[2017-03-08T09:33:36.392Z] [INFO] [#sid_101] [ContainerUtils] [marker:invoker_docker.unpause_start:23342565]
[2017-03-08T09:33:36.392Z] [INFO] [#sid_101] [SimpleExec] Running command: /usr/bin/docker unpause wsk0_177256_limeusibmcomactioninvokenodejs1488965337811001_20170308T092858748Z
[2017-03-08T09:33:36.407Z] [INFO] [#sid_101] [SimpleExec] Done running command: /usr/bin/docker unpause wsk0_177256_limeusibmcomactioninvokenodejs1488965337811001_20170308T092858748Z
[2017-03-08T09:33:36.407Z] [ERROR] [#sid_101] [ContainerUtils] stdout:  stderr: Error response from daemon: Container 480f96278f99592139a19e0cc4ba756f15d74c3052fa2d1524ebdf7edc840e70 is not paused [marker:invoker_docker.unpause_error:23342580:15]
[2017-03-08T09:33:36.407Z] [INFO] [#sid_101] [ContainerUtils] [marker:invoker_docker.rm_start:23342580]
[2017-03-08T09:33:36.407Z] [INFO] [#sid_101] [SimpleExec] Running command: /usr/bin/docker rm -f wsk0_177256_limeusibmcomactioninvokenodejs1488965337811001_20170308T092858748Z
```

## Additional information you deem important:

I suspect the culprit to be in the `killStragglers` function which is called during shutdown.
It directly calls `ContainerUtils.unpauseContainer` which implies `docker unpause`. It should however call `Container.unpause` on the container instance which knows whether to unpause using docker or runc.

**Comment by domdom82** [comment]
@perryibm ^^

**Comment by domdom82** [comment]
fixed by #1983

**Comment by rahulqelfo** [comment]
It seems it has come up again-->
`2018-07-06T14:36:48.239Z] [INFO] [#tid_34078] [InvokerReactive] recorded activation
[2018-07-06T14:36:48.245Z] [INFO] [#sid_102] [DockerClientWithFileAccess] running /usr/bin/docker pause 42d38c931b9f941054894e4522b32981303a2eea0f5a09d02e9688b055c0fead (timeout: 10 seconds) [marker:invoker_docker.pause_start:2538147]
[2018-07-06T14:36:48.393Z] [ERROR] [#sid_102] [DockerClientWithFileAccess] code: 137 , stdout: , stderr:  [marker:invoker_docker.pause_error:2538295:148]
[2018-07-06T14:36:48.393Z] [INFO] [#sid_102] [DockerClientWithFileAccess] running /usr/bin/docker rm -f 42d38c931b9f941054894e4522b32981303a2eea0f5a09d02e9688b055c0fead (timeout: 1 minute) [marker:invoker_docker.rm_start:2538295]
[2018-07-06T14:36:48.837Z] [INFO] [#sid_102] [DockerClientWithFileAccess]  [marker:invoker_docker.rm_finish:2538739:444]
[2018-07-06T14:46:42.577Z] [INFO] [#sid_102] [DockerClientWithFileAccess] running /usr/bin/docker unpause 5b66db0992fdd9558a63be60693f1de9584c9b27d12f7f60aaa070e1b991dd76 (timeout: 10 seconds) [marker:invoker_docker.unpause_start:3132478]
[2018-07-06T14:46:42.741Z] [ERROR] [#sid_102] [DockerClientWithFileAccess] code: 137 , stdout: , stderr:  [marker:invoker_docker.unpause_error:3132642:163]
`

Will provide more details soon.
